server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1183,24 +1183,27 @@ private UserVm upgradeStoppedVirtualMachine(Long vmId, Long svcOffId, Map<String
         // Check that the specified service offering ID is valid
         _itMgr.checkIfCanUpgrade(vmInstance, newServiceOffering);
 
-        DiskOfferingVO newROOTDiskOffering = _diskOfferingDao.findById(newServiceOffering.getId());
+        // Check if the new service offering can be applied to vm instance
```

Comment:
this comment could be a method name instead `canServiceOfferingBeAppliedToInstance(...)`